.github/workflows/build.yml

```diff
@@ -40,3 +40,42 @@ jobs:
         run: gem install --default -v0.11.0 uri
       - name: Run the tests
         run: bundle exec rake spec
+  run-specs-against-ruby-client-main:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    env:
+      BUNDLE_GEMFILE: gemfiles/ruby_client_main.gemfile
+      RSPEC_FULL_BACKTRACE: yes
+      CS_IDP_CLIENT_SECRET: ${{ secrets.CS_IDP_CLIENT_SECRET }}
+      CS_WORKSPACE: ${{ secrets.CS_WORKSPACE }}
+      CS_KMS_KEY_ARN: ${{ secrets.CS_KMS_KEY_ARN }}
+      CS_NAMING_KEY:  ${{ secrets.CS_NAMING_KEY }}
+      RAILS_VERSION: 6
+      ACTIVE_STASH_TEST_COLLECTION_PREFIX: run_${{ github.run_number }}.${{ github.run_attempt }}_job_${{ github.job }}_index_${{ strategy.job-index }}
+      PGDATABASE: activestash_test_run_${{ github.run_number }}.${{ github.run_attempt }}_job_${{ github.job }}_index_${{ strategy.job-index }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7
+          bundler-cache: false # don't use bundler-cache because we need to run bundle install manually in a later step
+      - uses: ankane/setup-postgres@v1
+        with:
+          database: ${{ env.PGDATABASE }}
+      # Installs GVB so that bundler can resolve the version of cipherstash-client when specified as a Git dependency
+      - name: "Install GVB"
+        run: gem install git-version-bump
```

Comment:
This is a bit clunky, but CI needs GVB available so that bundler can correctly resolve the version of ruby-client on main. 

Otherwise `bundle install` will fail with an error:
```
  Bundler could not find compatible versions for gem "cipherstash-client":
    In ruby_client_main.gemfile:
      cipherstash-client
  
      active_stash was resolved to 0.4.5, which depends on
        cipherstash-client (~> 0.12)
  
  Could not find gem 'cipherstash-client (~> 0.12)', which is required by gem
  'active_stash', in https://github.com/cipherstash/ruby-client (at main@b6f82af).
  
  The source contains the following gems matching 'cipherstash-client':
    * cipherstash-client-0.0.0.1.NOGVB
 ```

Example failure: https://github.com/cipherstash/activestash/runs/7685083887?check_suite_focus=true#step:3:46

Reply:
That's a fun little corner case in git-version-bump.  Pity transclusion never took off as a concept... in any event, I can't think of a better solution to the problem than what you've done here.  Nice one for figuring out what was going on.